### PR TITLE
Upgrade TypeScript to 5.2.2

### PR DESCRIPTION
**Pull request by ravicious**
https://devblogs.microsoft.com/typescript/announcing-typescript-5-2/

After the upgrade `yarn type-check` wasn't reporting any issues, but `yarn start-term` was returning some errors from Babel. I assumed it's a matter of updating `@babel/preset-typescript`, but that didn't help.

In the end, I just addressed the issues but I left `@babel/preset-typescript` upgraded because why not.

`tslib` was moved to the root package.json to be next to `typescript` dep.

### CODE REVIEW

File: web/packages/teleterm/src/ui/Search/searchResult.ts [hunk 1]
```diff
@@ -18,9 +18,6 @@ import type { ClusterUri } from 'teleterm/ui/uri';
 import type { Cluster } from 'teleterm/services/tshd/types';
 
 import type * as resourcesServiceTypes from 'teleterm/ui/services/resources';
-import type { SearchResultResource } from 'teleterm/ui/services/resources';
-
-export { SearchResultResource };
```
Comment (ravicious) [inline web/packages/teleterm/src/ui/Search/searchResult.ts:23]: This export was not used so I just removed it. Babel was complaining that `SearchResoultResource` was undefined.

File: package.json [hunk 2]
```diff
@@ -34,7 +34,8 @@
   },
   "devDependencies": {
     "prettier": "^2.5.0",
-    "typescript": "^5.1.3"
+    "tslib": "^2.6.2",
```
Comment (gzdunek) [inline package.json:37]: `tslib` should be moved to `dependencies` https://zenzes.me/today-i-learned-tslib-is-not-a-devdependency/

**Comment by gzdunek** [comment]
Could you also upgrade TS-eslint packages?

eslint is complaining about the incompatible TS version
```
$ eslint --quiet '+(e|web)/**/*.{ts,tsx,js,jsx}'
=============

WARNING: You are currently running a version of TypeScript which is not officially supported by @typescript-eslint/typescript-estree.

You may find that it works just fine, or you may not.

SUPPORTED TYPESCRIPT VERSIONS: >=3.3.1 <5.1.0

YOUR TYPESCRIPT VERSION: 5.2.2
```

I forgot to upgrade them when I bumped TS to 5.1.

File: package.json (hunk 2 above)
Comment (ravicious) [reply package.json]: I moved it back to the shared package then. It does get hoisted to some weird version though.

**Review by gzdunek — APPROVED** [review]

**Review by zmb3 — APPROVED** [review]

**Pull request merged**